pyfar/dsp/filter/fractional_octaves.py

```diff
@@ -329,6 +378,60 @@ def _coefficients_fractional_octave_bands(
     return sos
 
 
+def _check_fractional_octave_filter_tolerance(
+            filt, num_fractions, frequency_range, tolerance_class):
+    """
+    Check if pyfar.dsp.filter.fractional_octave_bands meet the tolerance
+    for the magnitude response returned by
+    pyfar.dsp.constants.fractional_octave_filter_tolerance.
+
+    Raises a ValueError, of the tolerance is not met.
+
+    Parameters
+    ----------
+    filt : FilterSOS
+        The fractional octave band filters.
+    num_fractions : int
+        The number of bands an octave is divided into. Eg., ``1`` refers to
+        octave bands and ``3`` to third octave bands.
+    frequency_range : array, tuple
+        The lower and upper frequency limits. The default is
+        ``frequency_range=(20, 20e3)``.
+    tolerance_class : int
+        The tolerance class as defined in the standard. Must be 1 or 2.
+    """
+
+    exact_center_frequencies = fractional_octave_frequencies(
+        num_fractions, frequency_range)[1]
+
+    ir = filt.impulse_response()
+    tf = pf.dsp.decibel(ir).squeeze()
+
+    # check the tolerance band by band
+    for n, center_frequeny in enumerate(exact_center_frequencies):
+
+        lower, upper, frequencies = \
+            pf.constants.fractional_octave_filter_tolerance(
+                center_frequeny, num_fractions, tolerance_class)
+
+        # find frequencies within the range where the tolerance is specified
+        f_mask = np.logical_and(ir.frequencies >= frequencies[0],
+                                ir.frequencies <= frequencies[-1])
+
+        # linearly interpolate the tolerance to the frequencies where the
+        # filters are given. This is fine, because the tolerance is a
+        # piece-wise linear function.
+        lower = np.interp(ir.frequencies[f_mask], frequencies, lower)
+        upper = np.interp(ir.frequencies[f_mask], frequencies, upper)
+
+        # check the current band
+        if np.any(tf[n, f_mask] < lower) or np.any(tf[n, f_mask] > upper):
+            raise ValueError(
+                f'Class {tolerance_class} tolerance not met. Increase the '
+                'filter order, decrease the tolerance class or disable the '
+                'check to solve this issue.')
```

Comment:
Continuing from the first comment:
... Maybe this can then be changed to a genral warning informing the users if the filter is class 1 and 2, only class 2 compliant or not at all.